packages/babel-plugin-transform-es2015-parameters/src/rest.js

```diff
@@ -106,10 +106,10 @@ export let visitor = {
     argsId._shadowedFunctionLiteral = path;
 
     function optimiseCandidate(parent, parentPath, offset) {
-      if (t.isReturnStatement(parentPath.parent) || t.isIdentifier(parentPath.parent.id)) {
+      if (t.isReturnStatement(parentPath.parent) || t.isIdentifier(parentPath.parent.id) || t.isIdentifier(parentPath.parent.left)) {
```

Comment:
What is `left` referring to here? Also note that you already have access to `parent` no need for `parentPath.parent`